### PR TITLE
feat: basic functionality

### PR DESCRIPTION
**Pull request by shady77**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Contributing
+
+The following is a set of guidelines for contributing to Ciklum digital organisation packages. Please spend several minutes in reading these guidelines before you create an issue or pull request.
+
+## Code of Conduct
+
+We have adopted a [Code of Conduct](https://github.com/ciklum-digital/module-skeleton/blob/master/CODE_OF_CONDUCT.md) that we expect project participants to adhere to. Please read the full text so that you can understand what actions will and will not be tolerated.
```
Comment (bopohasii) [inline CONTRIBUTING.md]: change url to waas

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+## Branch Organization
+
+There are three types of branch we maintain: `master` `issue/*` `feat/*`.
+ 
+- If you send a Fix Request, please do it in: `issue/short-issue-title`
+- If you send a Feature Request, please do it in: `feat/short-feature-title`
+
+## Reporting bugs and questions
+
+We are using [GitHub Issues](https://github.com/ciklum-digital/module-skeleton/issues) for bug tracing. Please, use [template](https://github.com/ciklum-digital/module-skeleton/tree/master/.github/ISSUE_TEMPLATE) for issue/feature creating.
```
Comment (bopohasii) [inline CONTRIBUTING.md]: change url to waas

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Before you reporting a bug, please make sure you've searched exists issues.
+
+## Sending a Pull Request
+
+The core team is monitoring for pull requests. We will review your pull request and either merge it, request changes to it, or close it with an explanation.
+
+**Before submitting a pull request**, please make sure the following is done:
+
+1. Before you reporting a bug, please make sure you've searched exists issues.
+1. Open a new issue in the [issues tracker](https://github.com/ciklum-digital/module-skeleton/issues)
```
Comment (bopohasii) [inline CONTRIBUTING.md]: change url to waas

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+1. Fork the repository and create your branch from [proper branch](#Branch-Organization).
+1. Run `yarn install` in the repository root.
+1. If you've fixed a bug or added code that should be tested, add tests!
+1. Ensure the test suite passes (yarn test). Tip: `yarn test -- --watch TestName` is helpful in development.
+1. Ensure the code lints passes. Run `yarn lint` for check it.
+1. We are using commitizen and commit lint. Use `yarn commit` for prompting commit message window.
+1. Submit a pull request, referencing any issues it addresses.
+1. For PR creation use [template]([template](https://github.com/ciklum-digital/module-skeleton/tree/master/.github/PULL_REQUEST_TEMPLATE.md))
```
Comment (bopohasii) [inline CONTRIBUTING.md]: change url to waas

File: package.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "waas",
+  "description": "Module skeleton | Start your module here",
```
Comment (bopohasii) [inline package.json]: change description

File: packages/angular/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# Change Log
```
Comment (bopohasii) [inline packages/angular/CHANGELOG.md]: to clear changelog

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# Change Log
```
Comment (bopohasii) [inline CHANGELOG.md]: to clear

File: packages/core/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# Change Log
```
Comment (bopohasii) [inline packages/core/CHANGELOG.md]: to clear

File: packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    const url = 'api/testing';
+    const headers = {Authorization: 'Basic'};
+    const params = {id: '15'};
+    const status = 200;
+    const statusText = 'Ok';
+
+    const data = {
+      method: 'GET', headers, params, url, before: (data1: IRequestConfigExecute) => data1, after: (data1) => data1
+    };
+    adapter.executeRequest(data)
```
Comment (bopohasii) [inline packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts]: move expect phase down

1. initial phase
2. exec phase
3. check phase

File: packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+    req.flush({data: 'some data'});
+    expect(req.request.method).toBe('GET');
+  });
+
+  it('should return response with meta', (done) => {
+    const url = 'api/testing';
+    const headers = {Authorization: 'Basic'};
+    const params = {id: '15'};
+    const status = 200;
+    const statusText = 'Ok';
```
Comment (bopohasii) [inline packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts]: Could your organize via mock files?
Global situation for each test case

File: packages/core/src/modules/waas/classes/waas-service-tree-node.class.ts [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+  hooks: IWaasHooks = { after: [], before: [] };
+
+  constructor(parent: WaasServiceTreeNode = null) {
+    super(parent);
+    this.parentNode = parent;
+  }
+
+  static isVariable(path: string): boolean {
+    return (path || '').length === 0 ? false : path[0] === ':';
+  }
```
Comment (bopohasii) [inline packages/core/src/modules/waas/classes/waas-service-tree-node.class.ts]: Is to complicated
static isVariable(path: string = ''): boolean {
         return path.charAt(0) === ':';
}
or path.indexOf(':') === 0

File: package.json (hunk 5 above)
Comment (shady77) [reply package.json]: done

File: CHANGELOG.md (hunk 7 above)
Comment (shady77) [reply CHANGELOG.md]: done

File: packages/core/CHANGELOG.md (hunk 8 above)
Comment (shady77) [reply packages/core/CHANGELOG.md]: done

File: packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts (hunk 9 above)
Comment (shady77) [reply packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts]: async test

File: packages/core/src/modules/waas/classes/waas-service-tree-node.class.ts (hunk 11 above)
Comment (shady77) [reply packages/core/src/modules/waas/classes/waas-service-tree-node.class.ts]: done

File: packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts (hunk 10 above)
Comment (shady77) [reply packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts]: done

**Pull request merged**